### PR TITLE
Release/1.0.1

### PR DESCRIPTION
**Pull request by cs-yash-patodkar**
Requirement : 
1. On selecting single module only modules with json fields should be in the dropdown
2. 'Title Should' be mandatory
3. 'Data Fields' Should be mandatory 
Changes:
1. Added a function to filter out modules with json field

### CODE REVIEW

File: widget/edit.controller.js [hunk 1]
```diff
@@ -16,6 +16,7 @@
     $scope.removeLayer = removeLayer;
     $scope.onChangeModuleType = onChangeModuleType;
     $scope.maxlayers = false;
+    $scope.customModuleList=[];
```
Comment (cs-suyog-jinturkar) [inline widget/edit.controller.js:19]: Why we are calling this as customModuleList? why not "JsonObjModuleList"?

File: widget/edit.html [hunk 2]
```diff
@@ -75,16 +75,20 @@ <h3 class="modal-title col-md-9">Edit Funnel Chart</h3> <button type="button" cl
                             data-form-name="'editFunnelWidgetForm'">
                         </div>
                     </div>
+                    <div id="customModuleField" 
+                    data-ng-class="{ 'has-error': editFunnelWidgetForm.customModuleField.$invalid && editFunnelWidgetForm.customModuleField.$touched }" >
+                        <label class="control-label" >Select Field
+                            <span class="text-danger">*</span>
+                        </label>
+                        <span data-uib-tooltip="Select the Field which has the data, only json type fields will be available in the dropdown"
```
Comment (cs-suyog-jinturkar) [inline widget/edit.html:83]: Can we keep all messages in a const var inside js and refer them here?

Comment (cs-suyog-jinturkar) [inline widget/edit.html:83]: Suggestion: Select the Field which has the data, only json type fields will be available in the dropdown --> Select a JSON type field

Comment (cs-yash-patodkar) [reply widget/edit.html:83]: Sure, but this is how all the other widgets are!

Comment (cs-yash-patodkar) [reply widget/edit.html:83]: How about :- "Select field which has relevant JSON object, to populate the funnel."?

**Pull request closed without merging**